python/lsst/obs/base/formatters/fitsExposure.py

```diff
@@ -516,8 +521,54 @@ class FitsExposureFormatter(FitsMaskedImageFormatter):
     type covariance violation ever becomes a practical problem.
     """
 
+    can_read_from_uri = True
     ReaderClass = ExposureFitsReader
 
+    def read_from_uri(self, uri: ResourcePath, component: str | None = None, expected_size: int = -1) -> Any:
+        # For now only support small non-pixel components. In future
+        # could work with cutouts.
+        if uri.isLocal:
+            # For a local URI allow afw to read it directly.
+            return NotImplemented
+        pixel_components = ("mask", "image", "variance")
+        if not component or component in pixel_components:
+            # For pixel access download the whole file.
+            return NotImplemented
+        try:
+            fs, fspath = uri.to_fsspec()
+            hdul = []
+            with fs.open(fspath) as f, astropy.io.fits.open(f) as fits_obj:
+                for hdu in fits_obj:
+                    hdr = hdu.header
+                    extname = hdr.get("EXTNAME")
+                    # Older files have IMAGE in EXTNAME in PRIMARY so check
+                    # for EXTEND=T.
+                    extend = hdr.get("EXTEND")
+                    if not extend and extname and extname.lower() in pixel_components:
+                        # Skip pixel data but retain header.
+                        stripped_hdu = astropy.io.fits.ImageHDU(
+                            data=np.zeros([1, 1], dtype=np.int32), header=hdr
+                        )
+                        hdul.append(stripped_hdu)
+                    else:
+                        hdul.append(hdu)
+                stripped_fits = astropy.io.fits.HDUList(hdus=hdul)
+                # Write the FITS file to in-memory FITS.
+                buffer = BytesIO()
+                stripped_fits.writeto(buffer)
+        except Exception:
+            # For some reason we can't open the remote file so fall back.
```

Comment:
I think we want a warning here, and if that's too noisy we should have a more specific `except` for the expected case and a warning for the fallback.

Reply:
Ah, ok, if there's already a warning that's fine.  Maybe a comment to that effect, then?

Reply:
I'm saying that if we give up on doing the partial read because of some strange fsspec/astropy configuration issue, then an attempt will be made to download the file and if the actual error is something like a permissions error then you will get an exception from that code path that will cause the formatter to raise.

Reply:
I'm primarily worried about future code bugs that make us silently fall back to the unoptimized code path.